src/WebJobs.Script/Workers/Rpc/Configuration/ProfileCondition.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.Rpc
+{
+    public enum ConditionType
+    {
+        [EnumMember(Value = "environment")]
+        Environment,
+        [EnumMember(Value = "hostProperty")]
+        HostProperty
+    }
+
+    public enum ConditionHostPropertyName
+    {
+        [EnumMember(Value = "sku")]
+        Sku,
+        [EnumMember(Value = "platform")]
+        Platform,
+        [EnumMember(Value = "hostVersion")]
+        HostVersion
+    }
+
+    public class ProfileCondition
+    {
+        /// <summary>
+        /// Gets or sets the type of the condition
+        /// </summary>
+        [JsonProperty(PropertyName = "type")]
+        public ConditionType Type { get; set; }
```

Comment:
This would ideally be the only property we care about for a condition implementation.

If you think about the interface for this (e.g. something like `IWorkerProfileCondition`), it would have a "Type" property (`string`), and a method for evaluation.